### PR TITLE
refactor: split trident.proto

### PR DESCRIPTION
**Pull request by yuanchaoa**
### This PR is for:

- Message

#### Checklist
- [ ] Added unit test.
#### Backport to branches
- main

### CODE REVIEW

File: message/agent.proto [hunk 1]
```diff
@@ -0,0 +1,775 @@
...
+    rpc Push(SyncRequest) returns (stream SyncResponse) {}
+    rpc Upgrade(UpgradeRequest) returns (stream UpgradeResponse) {}
+    rpc Query(NtpRequest) returns (NtpResponse) {}
+    rpc GenesisSync(GenesisSyncRequest) returns (GenesisSyncResponse) {}
+    rpc KubernetesAPISync(KubernetesAPISyncRequest) returns (KubernetesAPISyncResponse) {}
+    rpc GetKubernetesClusterID(KubernetesClusterIDRequest) returns (KubernetesClusterIDResponse) {}
+    rpc GPIDSync(GPIDSyncRequest) returns (GPIDSyncResponse) {}
+    rpc Plugin(PluginRequest) returns (stream PluginResponse) {}
+    // because gRPC cannot be initiated by server, the req/resp of this rpc is reversed
+    rpc RemoteExecute(stream RemoteExecResponse) returns (stream RemoteExecRequest) {}
```
Comment (yuanchaoa) [inline message/agent.proto:19]: @SongZhen0704 @jin-xiaofeng  @lzf575  这里删除了一些非agent使用的

File: message/agent.proto [hunk 2]
```diff
@@ -0,0 +1,775 @@
...
+
+message SkipInterface {
+    // 若该接口对应的虚拟机内已经部署采集器,
+    // 发送此接口给虚拟机所在宿主机采集器
+    optional uint64 mac = 1;
+}
+
+message SyncResponse {
+    optional Status status = 1;
+    optional string config = 2;
```
Comment (yuanchaoa) [inline message/agent.proto:346]: config 修改为传输yaml字符串  @sharang @jin-xiaofeng @SongZhen0704

File: message/agent.proto [hunk 3]
```diff
@@ -0,0 +1,775 @@
...
+}
+
+enum Status {
+    SUCCESS = 0;
+    FAILED = 1;
+    HEARTBEAT = 2;
+    CLUSTER_ID_NOT_FOUND = 10;
+}
+
+enum TapMode {
```
Comment (sharang) [inline message/agent.proto:124]: CaptureNetworkType

File: message/agent.proto [hunk 4]
```diff
@@ -0,0 +1,775 @@
...
+    optional uint64 version_platform_data = 9 [default = 0]; /* only platform data */
+    optional uint64 version_acls = 10 [default = 0];
+    optional uint64 version_groups = 11 [default = 0];
+    optional string current_k8s_image = 12;
+
+    optional string ctrl_ip = 21;
+    optional string host = 22;      // 表示hostname，操作系统的原始主机名，注册和信息同步使用
+    repeated string host_ips = 23;  // 仅作为注册使用
+    optional string ctrl_mac = 25;
+    optional string vtap_group_id_request = 26;  // 支持采集器自动加入组
```
Comment (sharang) [inline message/agent.proto:101]: agent_group_id_request

File: message/agent.proto [hunk 5]
```diff
@@ -0,0 +1,775 @@
...
+}
+
+enum CaptureSocketType {
+    AUTO = 0;
+    AF_PACKET_V1 = 1;
+    AF_PACKET_V2 = 2;
+    AF_PACKET_V3 = 3;
+}
+
+message TapType {
```
Comment (sharang) [inline message/agent.proto:165]: CaptureNetworkType

File: message/agent.proto [hunk 6]
```diff
@@ -0,0 +1,775 @@
...
+    optional uint32 tap_type = 1;
```
Comment (sharang) [inline message/agent.proto:166]: capture_network_type

File: message/agent.proto [hunk 7]
```diff
@@ -0,0 +1,775 @@
...
+    optional PacketType packet_type = 2;
+    optional uint32 vlan = 3;
+    optional string source_ip = 4;
+    optional uint32 tap_port = 5;
```
Comment (sharang) [inline message/agent.proto:170]: capture_network_port

File: message/agent.proto [hunk 8]
```diff
@@ -0,0 +1,775 @@
...
+    repeated Interface interfaces = 1;
+    repeated PeerConnection peer_connections = 3;
+    repeated Cidr cidrs = 4;
+}
+
+enum Action {
+    PACKET_CAPTURING = 1;  // 包存储（pcap）
+}
+
+enum TapSide {
```
Comment (sharang) [inline message/agent.proto:292]: ObservationSide

File: message/agent.proto [hunk 9]
```diff
@@ -0,0 +1,775 @@
...
+    ALL = 1;
+    FORWARD = 2;
+    BACKWARD = 3;
+}
+
+message NpbAction {
+    optional TunnelType tunnel_type = 1 [default = VXLAN];
+    optional uint32 tunnel_id = 2;
+    optional string tunnel_ip = 3;
+    optional TapSide tap_side = 4;
```
Comment (sharang) [inline message/agent.proto:315]: observation_side

File: message/agent.proto [hunk 10]
```diff
@@ -0,0 +1,775 @@
...
+    // =================================================================================
+    // 控制器向每个 Agent 下发一个 container list，其内容为该 Agent 运行操作
+    // 系统中的 container 列表（注意仅会下发本机的 container，不会包含其他机器的）
+    repeated Container containers = 8;
+    repeated Segment local_segments = 9;
+    repeated Segment remote_segments = 10;
+    optional bytes platform_data = 11;  // serialized result of `message PlatformData`, transmitted only when the content changes
+    optional bytes flow_acls = 12;      // serialized result of `message FlowAcls`, transmitted only when the content changes
+    optional bytes groups = 13;         // serialized result of `message Groups`, transmitted only when the content changes
+    repeated TapType tap_types = 14;
```
Comment (sharang) [inline message/agent.proto:366]: capture_network_types

File: message/agent.proto [hunk 11]
```diff
@@ -0,0 +1,775 @@
...
+    optional uint32 payload_slice = 5 [default = 65535];
+    optional uint32 npb_acl_group_id = 6;
+    optional uint32 tunnel_ip_id = 7;  // 分发点id, 限制在64000
+    optional Direction direction = 8 [default = ALL];
+}
+
+// 字段含义查看README
+message FlowAcl {
+    optional uint32 id = 1;
+    optional uint32 tap_type = 2;
```
Comment (sharang) [inline message/agent.proto:325]: capture_network_type

File: message/agent.proto [hunk 12]
```diff
@@ -0,0 +1,775 @@
...
+}
+
+message InterfaceInfo {
+    optional uint64 mac = 1;
+    // ip/masklen
+    repeated string ip = 2;
+    optional string name = 3;
+    optional string device_id = 4;
+
+    optional uint32 tap_index = 5;
```
Comment (sharang) [inline message/agent.proto:463]: capture_network_index ??

File: message/agent.proto [hunk 13]
```diff
@@ -0,0 +1,775 @@
...
+    repeated ProcessInfo process_entries = 5;
+}
+
+message GenesisSyncRequest {
+    optional uint64 version = 1 [default = 0];
+
+    optional AgentType agent_type = 2;
+
+    optional string source_ip = 3;
+    optional uint32 vtap_id = 4;
```
Comment (sharang) [inline message/agent.proto:500]: agent_id

File: message/agent.proto [hunk 14]
```diff
@@ -0,0 +1,775 @@
...
+
+message GenesisSyncResponse {
+    optional uint64 version = 1 [default = 0];
+}
+
+message KubernetesAPISyncRequest {
+    optional string cluster_id = 1;
+    optional uint64 version = 2;
+    optional string error_msg = 3;
+    optional uint32 vtap_id = 4;
```
Comment (sharang) [inline message/agent.proto:518]: agent_id

File: message/agent.proto [hunk 15]
```diff
@@ -0,0 +1,775 @@
...
+    //    field of the previous entry, then its field is assigned a value of 0.
+    //    When a field in Protobuf have a default value of 0, this allows us
+    //    to achieve very efficient compression.
+    COMPRESS_ALGO_IGNORE_IF_EQUAL_TO_PREVIOUS = 1;
+}
+
+message GPIDSyncRequest {
+    optional string ctrl_ip = 1;
+    optional string ctrl_mac = 2;
+    optional uint32 vtap_id = 3;
```
Comment (sharang) [inline message/agent.proto:599]: agent_id

File: message/agent.proto [hunk 16]
```diff
@@ -0,0 +1,775 @@
...
+    optional uint32 r_port = 4 [default = 0];
+    optional uint32 v_ipv4 = 5 [default = 0];
+    optional uint32 v_port = 6 [default = 0];
+}
+
+message RVData {
+    repeated RipToVip entries = 1;
+}
+
+message GPIDVTapData {
```
Comment (sharang) [inline message/agent.proto:667]: GPIDAgentData

**Comment by yuanchaoa** [comment]
修改合入到 #7695

**Pull request closed without merging**